Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -172,87 +175,101 @@ public void Serialize<T>(IIonWriter writer, T item)
                 return;
             }
 
-            if (item is bool)
+            var type = item.GetType();
+
+            if (type.IsAssignableTo(typeof(bool)))

```

Comment:
`is` is fine here because `bool` is a struct which cannot be extended

Reply:
Addressed.